packages/client/src/hooks/useTreasury.js

```diff
@@ -39,27 +47,43 @@ const doCreateTreasury = async (signerAddresses, threshold) => {
       arg(signerAddresses, t.Array(t.Address)),
       arg(threshold, t.UInt64),
     ],
-    limit: 80,
+    limit: CREATE_TREASURY_LIMIT,
   });
 };
 
 const doSendFlowToTreasury = async (treasuryAddr, amount) => {
   return await mutate({
     cadence: SEND_FLOW_TO_TREASURY,
     args: (arg, t) => [arg(treasuryAddr, t.Address), arg(amount, t.UFix64)],
-    limit: 55,
+    limit: REGULAR_LIMIT,
   });
 };
 
-const doProposeTransfer = async (treasuryAddr, recipientAddr, amount) => {
+const doProposeTransfer = async (
+  web3,
+  treasuryAddr,
+  recipientAddr,
+  amount
+) => {
+  const uFixAmount = String(parseFloat(amount).toFixed(8));
+  const tokenAddress = process.env.REACT_APP_FLOW_ENV === "emulator" ? "ee82856bf20e2aa6" : "9a0766d93b6608b7";
```

Comment:
is this for testing only?

Reply:
This action has an intent that contains the identifier of a recipient vault, which contains information about what account is the FungibleToken contract deployed to, so this is a "hacky" solution to sign the message in a correct way if we keep the intent as is.
TransferToken action should work for every type of FungibleToken that is available, so ideally intent should contain something readable, like $FLOW, $USDC, $FUSD... and this is also necessary for the UI when user proposes the action to transfer tokens from the treasury (to avoid hardcoding or changing the values that come from BE)

Reply:
got it. So we'll need to pass the token type based on user's selection and maintain an address mapping to be able to dynamically set here. Do we already have a ticket to cover this work? If no I can create one. cc @linxiao-li

Reply:
We can put the mapping information in `src/constants/maps.js`. I'll work with @KsenijaZivkovic to compose the map